File: Lesson_1/TapeEquilibrium/Python/Solution.py
# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"
import sys

def solution(A):
    # write your code in Python 2.7
    total_sum = sum(A)
    size = len(A)
    min_result = sys.maxint
    sum_left = 0
    for i in range (0, size-1):
        sum_left += A[i]
        result = abs(sum_left-(total_sum-sum_left))
        
        if result<min_result:
            min_result = result
    return min_result

File: Lesson_1/TapeEquilibrium/PHP/Solution.php
// you can write to stdout for debugging purposes, e.g.
// print "this is a debug message\n";

function solution($A) {
    // write your code in PHP5.5
    
    $summed_array = array_sum($A);
    $total_number = count($A);
    $left_summed = 0;
    $solution = PHP_INT_MAX;
    for($i = 0; $i < $total_number-1; ++$i)
    {
        $left_summed += $A[$i];
        $result = abs($left_summed - ($summed_array - $left_summed));
        if($result<$solution)
        {
            $solution = $result;
        }
    }
    
    return $solution;
}